pkg/assertions/yamlmeta.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package assertions
+
+import "github.com/vmware-tanzu/carvel-ytt/pkg/yamlmeta"
+
+// AddValidations appends validation Rules to node's validations metadata, later retrieved via GetValidations().
+func AddValidations(node yamlmeta.Node, rules []Rule) {
+	metas := node.GetMeta("validations")
+	if currRules, ok := metas.([]Rule); ok {
```

Comment:
I see how `ok` is `false` when `metas` is `nil`. That's pretty slick.

At the same time, there's the (should never happen) situation where `metas` is both _not `nil`_ and _not `[]Rule`_.

Given the role that `ytt` plays — a often low-level integrated piece in a larger mass of infrastructure automation — it pays to be _as boring as possible_ in order to avoid any chance of strange behaviors or unexpected errors.
For me, it took a bit to parse all of what's intended here, given that a failed type assertion is playing the role of a nil-check, here.

Suggestion: for readability and reliability's sake, let's err on the side of "boring code":
- split the lazy initialization (i.e. nil check) from the type assertion.
- on failed type assertion, panic.

_(I clearly didn't read through this closely enough, prior. Apologies for the late call, here.)_